website/blog/2023-02-01-technical-preview/index.mdx

```diff
@@ -0,0 +1,113 @@
+
+```mermaid
+---
+title: glasskube install [package]
+---
+flowchart BT
+  UI([UI])-- via local server<br>http://localhost:8580 ---Client(Client)
+  CLI([CLI])-- cobra cli ---Client
+  Client-- 1. validate package -->Repo[(Public Glasskube<br>Package Repo)]
+  Client-- 2. create<br>`Package` CR -->Kubernetes(((Kubernetes API)))
+  subgraph Cluster
+    Kubernetes-- 3. reconcile<br>`Package` -->PackageController
+    PackageController-- 4. create `PackageInfo`<br>if not present-->Kubernetes
+    Kubernetes-- 5. reconcile<br>`PackageInfo`-->PackageInfoController
+    end
+  PackageInfoController<-- 6. update package manifest -->Repo
+  subgraph Cluster
+    PackageInfoController-- 7. update manifest<br>in `PackageInfo` -->Kubernetes
+    Kubernetes-- 8. reconcile<br>`PackageInfo` -->PackageController
+    PackageController-- 9. deploy package -->Kubernetes
+  end
+
+  Kubernetes-- 10. package status -->Client
+```
+
+### Challenges that need to be solved
+
+We already covered some our next features in our [public roadmap](https://glasskube.dev/roadmap), but I also like to shortly speak about broader challenges.
+
+#### Kubernetes version compatibility
+
+Kubernetes releases minor versions every 4 months which often comes with new api versions. So package authors need to adapt their packages to the new api versions.
+Kubernetes is designed in a way that often multiple api versions are implemented in specific release of Kubernetes, so packages often work with a broader range
+of Kubernetes versions.
+These compatible versions are often only documented in the package distributors' changelog.
+Glasskube aims to incorporate this kind of metadata in combination with automatic checks from tools like [kube-no-trouble](https://github.com/doitintl/kube-no-trouble)
+or [Pluto](https://github.com/FairwindsOps/pluto).
+
+The user should not need to manually check all packages for compatability as well as package developers should get feedback if they are not compatible with the latest api versions.
+
+#### Package dependencies
+
+Cloud native applications often work together as there are already quite a few packages that are present in most clusters anyway - like cert-manager, Ingress controller or
+database operators. Due to the lack of a package manager and ecosystem these dependencies are still often only documented in the Getting started section of an application.
+
+In an ideal world a developer can just refer to a dependency as a dependency and the package manager makes sure that all these prerequisites are fulfilled.
+
+#### Testing
+
+In order to support multiple kubernetes versions, dependencies and packages Glasskube need to build massive automated testing infrastructure for all packages in its central package repository.
+
+
+#### Feedback and package quality
+
+Similar to other package managers user feedback is highly important to share problems and work together on solutions.
```

Comment:
I don't know what to make of this sentence. What does a package manager have to do with user feedback being important? And why does user feedback help with working together, as opposed to working alone?